src/pages/interface/AssociateToken.vue

```diff
@@ -9,66 +9,109 @@
-      </label>
-
-      <EntityIdInput
-        v-model="state.token"
-        data-cy-token-id
-        type="token"
-        :placeholder="$t('InterfaceToolTile.associateToken.placeholder')"
-      />
+  <div class="p-8 mt-10">
+    <div
+      class="transition-all duration-300 mb-8 w-full md:w-3/4 md:m-auto text-center"
+    > 
+      <div class="text-left">
+        <label class="font-medium text-squant dark:text-argent">
+          {{ $t("InterfaceToolTile.associateToken.label.tokenID") }}
+        </label>
+        <EntityIdInput
+          v-model="state.token"
+          data-cy-token-id
+          type="token"
+          :placeholder="$t('InterfaceToolTile.associateToken.placeholder')"
+        />
+      </div>
+
+      <Button
+        color="green"
+        class="mt-8 md:mb-2 mb-12 p-3 w-3/4 m-auto"
+        :disabled="!state.token"
+        @click="addToken"
+      >
+        {{ $t("InterfaceAssociateToken.button.add.token") }}
+      </Button>
 
       <InputError
         v-if="state.errorMessage.length > 0"
         data-cy-input-error
       >
         {{ state.errorMessage }}
       </InputError>
+
+      <div
+        class="rounded p-8 mt-12 bg-opacity-50 bg-first-snow dark:text-white dark:bg-midnight-express min-h-[200px] transition-all duration-300 overflow-auto"
+        :class="{
+          'opacity-100': state.tokens.length > 1,
+          'opacity-0 invisible -mt-44 h-0': state.tokens.length == 0
+        }"
+      >
+        <div
+          v-for="(token, i) in state.tokens"
+          :key="token.toString()"
+          class="m-2 p-6 rounded bg-white hover:bg-first-snow shadow-md dark:shadow-none dark:hover:bg-black-out dark:bg-ruined-smores text-left"
+        >
+          <TokenItem
+            :token-id="token.toString()"
+            :index="i"
+            @remove="remove"
+          />
+        </div>
+      </div>
     </div>
+  </div>
 
-    <div class="mt-48 border-b border-cerebral-grey" />
+  <div class="mt-48 border-b border-cerebral-grey" />
 
-    <div class="flex flex-col items-center max-w-lg m-auto">
-      <Button
-        data-cy-submit
-        type="submit"
-        color="green"
-        class="w-full p-4 mt-10"
-        :disabled="state.token == null"
-      >
-        {{ $t("InterfaceToolTile.associateToken.title") }}
-      </Button>
+  <div class="flex flex-col items-center max-w-lg m-auto">
+    <Button
+      data-cy-submit
+      type="submit"
+      color="green"
+      class="w-full p-4 mt-10"
+      :disabled="state.tokens.length == 0"
+      @click="handleAssociate"
+    >
+      {{ associateTokenButton }}
+    </Button>
 
-      <Button
-        color="white"
-        :to="{ name: 'tools' }"
-        class="p-2 mt-4 w-36"
-      >
-        {{ $t("BaseButton.cancel") }}
-      </Button>
-    </div>
-  </form>
+    <Button
+      color="white"
+      :to="{ name: 'tools' }"
+      class="p-2 mt-4 w-36"
+      @click="handleAssociate"
```

Comment:
put the `form` back and you won't need to worry about handleAssociate on different buttons. Make each ending button a button of type = "submit"